### ISSUE TITLE
Use rappdirs for saving temporary files

### ISSUE THREAD

**Issue by jdblischak** [workflowr/workflowr#86]
For [saving log files](https://github.com/jdblischak/workflowr/blob/7242a340986fa518b7f35e15de488d2853ea587d/R/wflow_build.R#L150) and creating temporary directories for unit tests, I  am using `/tmp`. The advantage of this approach over using `tempdir()` is that 1) the filepath is easier to read and type into a Terminal, and 2) it persists after the current R session ends. But of course the main downside is that `/tmp` doesn't really exist on Windows, and thus making the code work cross-platform is more difficult. From `?tempdir`:

```
tempdir() # works on all platforms with a platform-dependent result
```

To get this to work, I've had to write some [uncomfortably hacky code](https://github.com/jdblischak/workflowr/blob/7242a340986fa518b7f35e15de488d2853ea587d/tests/testthat/test-wflow_start.R#L26).

This has been on my to-do list to fix, but I recently found a bigger problem with this approach. When Joyce or I run workflowr on the same partition on Midway, the directory `/tmp/workflowr` is owned by one of us. Thus when `wflow_build()` tries to create the log files and `/tmp/workflowr` is owned by the other person, it fails.

My plan is to use [rappdirs](https://cran.r-project.org/package=rappdirs). It finds the path for saving temporary files for each operating system. I can save the log files in `user_log_dir()` and the temporary directories for unit testing in `user_data_dir()`. As an example on Midway:

```
> library("rappdirs")
> user_cache_dir(appname = "workflowr")
[1] "/home/jdblischak/.cache/workflowr"
> user_config_dir(appname = "workflowr")
[1] "/home/jdblischak/.config/workflowr"
> user_data_dir(appname = "workflowr")
[1] "/home/jdblischak/.local/share/workflowr"
> user_log_dir(appname = "workflowr")
[1] "/home/jdblischak/.cache/workflowr/log"
```

Note that I'll need to also need to specify the argument `appauthor = Sys.info()["user"]`, which is used on Windows and safely ignored on Unix-alike systems.

**Comment by pcarbo** [comment]
👍

**Comment by jdblischak** [comment]
Well actually this approach may not work due to [CRAN policies](https://cran.r-project.org/web/packages/policies.html):

> - Packages should not write in the users’ home filespace, nor anywhere else on the file system apart from the R session’s temporary directory

This is super frustrating. Using `/tmp` on Windows is causing problems, but I remembered why I started specifying `/tmp` in the first place: `tempfile()` on macOS uses symlinks that were causing problems: https://github.com/jdblischak/workflowr/commit/f709fad01f95eb671fae70728ffb82453abda1ab

I'll have to re-investigate the macOS issue. I need a cross-platform solution for writing temporary files, and I'd prefer to not have to re-implement `tempfile()`.
